Tests.Unit.Presentation.Web/Services/OrganizationServiceTest.cs

```diff
@@ -260,21 +263,23 @@ public void RemoveUser_Returns_Ok()
             var organization = new Organization();
             ExpectGetOrganizationByKeyReturns(organizationId, organization);
             ExpectAllowModifyReturns(organization, true);
+            ExpectTransactionBeginReturns();
             var matchedRight1 = CreateRight(organizationId, userId);
             var matchedRight2 = CreateRight(organizationId, userId);
             var unmatchedRight1 = CreateRight(A<int>(), userId);
             var unmatchedRight2 = CreateRight(organizationId, A<int>());
             _orgRightRepository.Setup(x => x.AsQueryable()).Returns(new[] { matchedRight1, unmatchedRight1, matchedRight2, unmatchedRight2 }.AsQueryable());
+            ExpectOrganizationRightsRemoveRoleReturnsSuccess();
 
             //Act
             var result = _sut.RemoveUser(organizationId, userId);
 
             //Assert that only the right entities were removed
             Assert.True(result.Ok);
-            _orgRightRepository.Verify(x => x.DeleteByKey(matchedRight1.Id), Times.Once);
-            _orgRightRepository.Verify(x => x.DeleteByKey(matchedRight2.Id), Times.Once);
-            _orgRightRepository.Verify(x => x.DeleteByKey(unmatchedRight1.Id), Times.Never);
-            _orgRightRepository.Verify(x => x.DeleteByKey(unmatchedRight2.Id), Times.Never);
+            _organizationRightsServiceMock.Verify(x => x.RemoveRole(matchedRight1.Id), Times.Once);
+            _organizationRightsServiceMock.Verify(x => x.RemoveRole(matchedRight2.Id), Times.Once);
```

Comment:
missing a unit test for the domain event handler... it does a lot of stuff that we want to test